hail/python/hail/backend/backend.py

```diff
@@ -88,11 +87,11 @@ def fs(self):
 
 
 class SparkBackend(Backend):
-    def __init__(self):
-        self._fs = HadoopFS()
-
     @property
     def fs(self):
+        if not hasattr(self, '_fs'):
```

Comment:
I prefer to set `self._fs = None` in `__init__` and check if it is None, rather than checking if the attribute is defined.  In particular, lint tools find it easier to make sense of.